plugins/plugin-svn/che-plugin-svn-ext-ide/src/main/java/org/eclipse/che/plugin/svn/ide/common/SubversionActionPresenter.java

```diff
@@ -108,10 +137,56 @@ protected void printResponse(final String command, final List<String> output, fi
     }
 
     /**
+     * Performs subversion operation. If this operations fails with authorization error
+     * the operation wil be recalled with requested credentials
+     *
+     * @param notification
+     *         progress notification that was notified when operation started
+     */
+    protected <Y> Promise<Y> performOperationWithCredentialsRequestIfNeeded(final SVNOperation<Y> operation,
+                                                                            @Nullable final StatusNotification notification) {
+        return operation.perform(new Credentials()).catchErrorPromise(new Function<PromiseError, Promise<Y>>() {
+            @Override
+            public Promise<Y> apply(PromiseError error) throws FunctionException {
+                if (getErrorCode(error.getCause()) == ErrorCodes.UNAUTHORIZED_SVN_OPERATION) {
+                    if (notification != null) {
+                        notification.setTitle(error.getMessage());
+                        notification.setStatus(FAIL);
+                    } else {
+                        notificationManager.notify(error.getMessage(), FAIL, FLOAT_MODE);
+                    }
+
+                    return credentialsDialog.askCredentials().catchError(new Operation<PromiseError>() {
+                        @Override
+                        public void apply(PromiseError error) throws OperationException {
+                            notificationManager.notify(error.getMessage(), FAIL, FLOAT_MODE);
+                        }
+                    }).thenPromise(new Function<Credentials, Promise<Y>>() {
+                        @Override
+                        public Promise<Y> apply(Credentials credentials) throws FunctionException {
+                            return operation.perform(credentials);
+                        }
+                    });
+                }
+                return Promises.reject(error);
+            }
+        });
+    }
+
+    /**
+     * Subversion operation.
+     */
+    protected interface SVNOperation<Y> {
```

Comment:
Let's use Subversion word instead of SVN to do in the same way to other names. Also please think about adding to interface name something that will say about that this operation may require authorization to wit credentials